### PR TITLE
Test case to verify StorageClassClaim

### PR DESCRIPTION
**Pull request by jilju**
Test case to verify StorageClassClaim.
Added function to create StorageClassClaim.
Added StorageClassClaim template yaml file.
Added StorageClassClaim kind in constants.py
Added class StorageClassClaim.

Signed-off-by: Jilju Joy <jijoy@redhat.com>

### CODE REVIEW

**Review by fbalak — COMMENTED** [review]
I suggest to create a python class for storageclassclaim and put it into ocs_ci/ocs/resources.

**Comment by jilju** [comment]
> I suggest to create a python class for storageclassclaim and put it into ocs_ci/ocs/resources.

@fbalak 
Created class for storageclassclaim resource

File: ocs_ci/helpers/helpers.py [hunk 1]
```diff
@@ -3909,3 +3909,46 @@ def get_noobaa_db_used_space():
         f"noobaa_db used space is {df_out[-4]} which is {df_out[-2]} of the total PVC size"
     )
     return df_out[-4]
+
+
+def create_storageclassclaim(
```
Comment (fbalak) [inline ocs_ci/helpers/helpers.py:3914]: Please move the function to [ocs_ci/ocs/resources/storageclassclaim.py](https://github.com/red-hat-storage/ocs-ci/pull/5935/files#diff-deff0564df2356f1bee617e0d3f5280037478ea7837236a19d4023fde20b5fac). This is directly linked to the resource module and shouldn't be in helpers.

Comment (jilju) [reply ocs_ci/helpers/helpers.py]: @fbalak Moved the function to storageclassclaim.py

**Review by fbalak — APPROVED** [review]

**Comment by jilju** [comment]
Verified: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/691/
The test failed in the last step due to some issue which is not related to this specific tests case.
Leftover check failure is due to the bug https://bugzilla.redhat.com/show_bug.cgi?id=2123697

**Comment by jilju** [comment]
Verified  locally using ignore_leftovers marker to avoid hitting the issue https://bugzilla.redhat.com/show_bug.cgi?id=2123697


```
$ run-ci multicluster 2 tests/managed-service/test_storageclassclaim.py --ocsci-conf=conf/ocsci/disable_pagerduty.yaml --cluster1 --cluster-name jijoy-sep2-c1 --cluster-path ~/ms-consumer/ --ocp-version 4.10 --ocs-version 4.11 --ocsci-conf=conf/deployment/rosa/managed_3az_consumer_qe_3m_3w_m52x.yaml --cluster2 --cluster-name jijoy-sep2-pr --cluster-path ~/ms-provider/ --ocp-version 4.10 --ocs-version 4.11 --ocsci-conf=conf/deployment/rosa/managed_3az_provider_qe_3m_3w_m54x.yaml


..................
..................

======================================================================================== test session starts =========================================================================================
platform linux -- Python 3.8.13, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/jijoy/ocs-ci, configfile: pytest.ini
plugins: html-3.1.1, flaky-3.7.0, marker-bugzilla-0.9.4, repeat-0.9.1, metadata-1.11.0, logger-0.5.1, ordering-0.6
collected 1 item                                                                                                                                                                                     

tests/managed-service/test_storageclassclaim.py::TestStorageClassClaim::test_verify_storageclassclaim 
------------------------------------------------------------------------------------------- live log setup -------------------------------------------------------------------------------------------
22:17:12 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - testrun_name: OCS4-11-Downstream-OCP4-11-ROSA-MANAGED-1AZ-RHCOS-3M-3W
........................
........................
------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------
22:18:48 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc new-project namespace-test-4a56ce65c5514c1fa38505b17
22:18:50 - MainThread - ocs_ci.ocs.resources.ocs - INFO - C[jijoy-sep2-c1] - Adding StorageClassClaim with name storageclassclaim-test-cephblockpool-7e6
22:18:50 - MainThread - ocs_ci.utility.templating - INFO - C[jijoy-sep2-c1] - apiVersion: ocs.openshift.io/v1alpha1
kind: StorageClassClaim
metadata:
  name: storageclassclaim-test-cephblockpool-7e6
  namespace: namespace-test-4a56ce65c5514c1fa38505b17
spec:
  type: blockpool

22:18:50 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 create -f /tmp/StorageClassClaim8w40_778 -o yaml
22:18:51 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClassClaim storageclassclaim-test-cephblockpool-7e6 -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml
22:18:51 - MainThread - test_storageclassclaim - INFO - C[jijoy-sep2-c1] - Creating storageclassclaims
22:18:51 - MainThread - ocs_ci.ocs.resources.ocs - INFO - C[jijoy-sep2-c1] - Adding StorageClassClaim with name storageclassclaim-test-cephblockpool-88b
22:18:51 - MainThread - ocs_ci.utility.templating - INFO - C[jijoy-sep2-c1] - apiVersion: ocs.openshift.io/v1alpha1
kind: StorageClassClaim
metadata:
  name: storageclassclaim-test-cephblockpool-88b
  namespace: namespace-test-4a56ce65c5514c1fa38505b17
spec:
  type: blockpool

22:18:51 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 create -f /tmp/StorageClassClaimc6ktc69j -o yaml
................................
...............................
22:19:17 - MainThread - ocs_ci.ocs.ocp - INFO - C[jijoy-sep2-c1] - status of storageclassclaim-test-cephblockpool-88b at PHASE reached condition!
22:19:17 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClass storageclassclaim-test-cephblockpool-7e6 -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml
22:19:18 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClass storageclassclaim-test-cephblockpool-88b -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml
22:19:19 - MainThread - test_storageclassclaim - INFO - C[jijoy-sep2-c1] - Creating PVCs and pods
22:19:19 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc new-project namespace-test-031b59a0901b4f2599937528a
22:19:20 - MainThread - ocs_ci.ocs.resources.ocs - INFO - C[jijoy-sep2-c1] - Adding PersistentVolumeClaim with name pvc-test-713e0a06cddf41debab53ab1828393c
22:19:20 - MainThread - ocs_ci.utility.templating - INFO - C[jijoy-sep2-c1] - apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc-test-713e0a06cddf41debab53ab1828393c
  namespace: namespace-test-031b59a0901b4f2599937528a
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
  storageClassName: storageclassclaim-test-cephblockpool-7e6

22:19:20 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-031b59a0901b4f2599937528a create -f /tmp/PersistentVolumeClaimrrmiq0_u -o yaml
22:19:21 - MainThread - ocs_ci.ocs.resources.ocs - INFO - C[jijoy-sep2-c1] - Adding PersistentVolumeClaim with name pvc-test-01a89554cd3841f48bfa6451c2895dc
22:19:21 - MainThread - ocs_ci.utility.templating - INFO - C[jijoy-sep2-c1] - apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc-test-01a89554cd3841f48bfa6451c2895dc
  namespace: namespace-test-031b59a0901b4f2599937528a
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
  storageClassName: storageclassclaim-test-cephblockpool-7e6
  volumeMode: Block

22:19:21 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-031b59a0901b4f2599937528a create -f /tmp/PersistentVolumeClaimq7pbrphy -o yaml
22:19:22 - MainThread - ocs_ci.ocs.resources.ocs - INFO - C[jijoy-sep2-c1] - Adding PersistentVolumeClaim with name pvc-test-83fbbd3ad29e43e28055be4ac952ff6
22:19:22 - MainThread - ocs_ci.utility.templating - INFO - C[jijoy-sep2-c1] - apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc-test-83fbbd3ad29e43e28055be4ac952ff6
  namespace: namespace-test-031b59a0901b4f2599937528a
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: 3Gi
  storageClassName: storageclassclaim-test-cephblockpool-7e6
  volumeMode: Block

22:19:22 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-031b59a0901b4f2599937528a create -f /tmp/PersistentVolumeClaimf2_h91n9 -o yaml
22:19:23 - MainThread - ocs_ci.ocs.ocp - INFO - C[jijoy-sep2-c1] - Waiting for a resource(s) of kind PersistentVolumeClaim identified by name 'pvc-test-713e0a06cddf41debab53ab1828393c' using selector None at column name STATUS to reach desired condition Bound
............................
...........................
22:19:35 - MainThread - ocs_ci.ocs.resources.ocs - INFO - C[jijoy-sep2-c1] - Adding Pod with name pod-test-rbd-458c6834f0994314b22cfabe910
22:19:35 - MainThread - ocs_ci.utility.templating - INFO - C[jijoy-sep2-c1] - apiVersion: v1
kind: Pod
metadata:
  name: pod-test-rbd-458c6834f0994314b22cfabe910
  namespace: namespace-test-031b59a0901b4f2599937528a
spec:
  containers:
  - image: quay.io/ocsci/nginx:latest
    name: web-server
    volumeMounts:
    - mountPath: /var/lib/www/html
      name: mypvc
  volumes:
  - name: mypvc
    persistentVolumeClaim:
      claimName: pvc-test-713e0a06cddf41debab53ab1828393c
      readOnly: false

22:19:35 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-031b59a0901b4f2599937528a create -f /tmp/POD_nsd_ssiu -o yaml
..........................
..........................
..........................
22:23:58 - MainThread - test_storageclassclaim - INFO - C[jijoy-sep2-c1] - IO finished on pod pod-test-cephfs-d52074e59021464aa7101c4b
PASSED
----------------------------------------------------------------------------------------- live log teardown ------------------------------------------------------------------------------------------
22:23:58 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-031b59a0901b4f2599937528a delete Pod pod-test-rbd-458c6834f0994314b22cfabe910
.....................................
.....................................

22:24:51 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 delete StorageClassClaim storageclassclaim-test-cephblockpool-88b
22:24:52 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClassClaim storageclassclaim-test-cephblockpool-88b -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml
22:24:53 - MainThread - ocs_ci.utility.utils - WARNING - C[jijoy-sep2-c1] - Command stderr: Error from server (NotFound): storageclassclaims.ocs.openshift.io "storageclassclaim-test-cephblockpool-88b" not found

22:24:53 - MainThread - ocs_ci.ocs.ocp - WARNING - C[jijoy-sep2-c1] - Failed to get resource: storageclassclaim-test-cephblockpool-88b of kind: StorageClassClaim, selector: None, Error: Error during execution of command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClassClaim storageclassclaim-test-cephblockpool-88b -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml.
Error is Error from server (NotFound): storageclassclaims.ocs.openshift.io "storageclassclaim-test-cephblockpool-88b" not found

22:24:53 - MainThread - ocs_ci.ocs.ocp - WARNING - C[jijoy-sep2-c1] - Number of attempts to get resource reached!
22:24:53 - MainThread - ocs_ci.ocs.ocp - INFO - C[jijoy-sep2-c1] - StorageClassClaim storageclassclaim-test-cephblockpool-88b got deleted successfully
22:24:53 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 delete StorageClassClaim storageclassclaim-test-cephblockpool-7e6
22:24:54 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClassClaim storageclassclaim-test-cephblockpool-7e6 -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml
22:24:55 - MainThread - ocs_ci.utility.utils - WARNING - C[jijoy-sep2-c1] - Command stderr: Error from server (NotFound): storageclassclaims.ocs.openshift.io "storageclassclaim-test-cephblockpool-7e6" not found

22:24:55 - MainThread - ocs_ci.ocs.ocp - WARNING - C[jijoy-sep2-c1] - Failed to get resource: storageclassclaim-test-cephblockpool-7e6 of kind: StorageClassClaim, selector: None, Error: Error during execution of command: oc -n namespace-test-4a56ce65c5514c1fa38505b17 get StorageClassClaim storageclassclaim-test-cephblockpool-7e6 -n namespace-test-4a56ce65c5514c1fa38505b17 -o yaml.
Error is Error from server (NotFound): storageclassclaims.ocs.openshift.io "storageclassclaim-test-cephblockpool-7e6" not found

22:24:55 - MainThread - ocs_ci.ocs.ocp - WARNING - C[jijoy-sep2-c1] - Number of attempts to get resource reached!
22:24:55 - MainThread - ocs_ci.ocs.ocp - INFO - C[jijoy-sep2-c1] - StorageClassClaim storageclassclaim-test-cephblockpool-7e6 got deleted successfully
..........................
.........................
22:25:47 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Executing command: oc -n openshift-storage exec rook-ceph-tools-74cff48f58-smdmc -- ceph health
22:25:50 - MainThread - ocs_ci.utility.utils - INFO - C[jijoy-sep2-c1] - Ceph cluster health is HEALTH_OK.
22:25:50 - MainThread - tests.conftest - INFO - C[jijoy-sep2-c1] - Ceph health check passed at teardown


=================================================================================== 1 passed in 518.31s (0:08:38) ====================================================================================



```

**Review by fbalak — DISMISSED** [review]

File: tests/managed-service/test_storageclassclaim.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import ManageTest, tier1, polarion_id
+from ocs_ci.ocs.resources.storageclassclaim import create_storageclassclaim
+from ocs_ci.helpers.helpers import create_ocs_object_from_kind_and_name
+
+log = logging.getLogger(__name__)
+
+
+@tier1
```
Comment (ebondare) [inline tests/managed-service/test_storageclassclaim.py:11]: I'm a bit confused: if storageclassclaim test is relevant for managed ocs only, I'd expect a marker 'skip if not managed service'. If it's more generic, I'd expect it not to be in managed_service folder.

Comment (jilju) [reply tests/managed-service/test_storageclassclaim.py:16]: This is relevant for MS only. Added 'ms_consumer_required' marker.

**Review by fbalak — APPROVED** [review]

**Review by ebondare — APPROVED** [review]
LGTM

**Pull request merged**